core/src/test/java/com/expediagroup/streamplatform/streamregistry/core/services/StreamServiceTest.java

```diff
@@ -182,6 +183,35 @@ public void update() {
     verify(streamRepository).saveSpecification(entity);
   }
 
+  @Test(expected = ValidationException.class)
+  public void updateWithChangedSchemaKey() {
+    StreamKey key = new StreamKey();
+    key.setDomain("domain");
+    key.setName("stream");
+    key.setVersion(1);
+    Stream existingEntity = new Stream(key, new Specification(), new SchemaKey("domain", "stream_v1"));
+    Stream updatedEntity = new Stream(key, new Specification(), new SchemaKey("domain", "stream_v2"));
+    when(streamRepository.findById(key)).thenReturn(Optional.of(existingEntity));
+    streamService.update(updatedEntity);
+    verify(streamRepository).findById(key);
+  }
+
+  @Test
+  public void updateWithSchemaKeyNull() {
+    StreamKey key = new StreamKey();
+    key.setDomain("domain");
+    key.setName("stream");
+    key.setVersion(1);
+    Stream existingEntity = new Stream(key, new Specification(), new SchemaKey("domain", "stream_v1"));
+    Stream updatedEntity = new Stream(key, new Specification(), null);
+    when(streamRepository.findById(key)).thenReturn(Optional.of(existingEntity));
+    streamService.update(updatedEntity);
```

Comment:
can we validate the output contains the non null schemakey

Reply:
added